src/lib/components/shared/vercel_hover.svelte

```diff
@@ -0,0 +1,70 @@
+<script lang="ts">
+    export let glider_container_class = "";
+    export let active_element_class = "";
+    export let direction: "horizontal" | "vertical";
+    /** vercel effect */
+    let hover_glider_element: HTMLElement,
+        glider_container_element: HTMLElement,
+        GLIDER_TRANSITION_DURATION = 200,
+        is_hovered = false,
+        mouse_leave_timeout: NodeJS.Timer | null = null;
+
+    const handle_mouseenter = (event: MouseEvent) => {
+        const target = event.target as HTMLElement;
+
+        // Do some magic here to get the target's height and width
+        // Don't change the position of this code.
+        // It will cause animation jank
+        hover_glider_element.style.height = getComputedStyle(target).height;
+        hover_glider_element.style.width = getComputedStyle(target).width;
+
+        const target_z_index = getComputedStyle(target).zIndex;
+
+        if (target_z_index === "auto") {
+            target.style.zIndex = "0";
+        }
+        hover_glider_element.style.zIndex = String(Number(getComputedStyle(target).zIndex) - 1);
+
+        switch (direction) {
+            case "vertical":
+                hover_glider_element.style.transform = `translateY(${target.offsetTop}px)`;
+                break;
+            case "horizontal":
+                hover_glider_element.style.transform = `translateX(${target.offsetLeft - glider_container_element.offsetLeft}px)`;
```

Comment:
I dont understand as to why this could be the case : 
`target.offsetLeft - glider_container_element.offsetLeft`

This should have been `target.offsetLeft`